tensorflow-training/src/main/java/org/tensorflow/training/optimizers/AdaDelta.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2019, Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.tensorflow.training.optimizers;
+
+import org.tensorflow.Graph;
+import org.tensorflow.Operand;
+import org.tensorflow.Output;
+import org.tensorflow.op.core.Variable;
+import org.tensorflow.types.TFloat32;
+import org.tensorflow.types.family.TType;
+
+import java.util.List;
+
+/**
+ * Optimizer that implements the Adadelta algorithm.
+ *
+ * See the <a href="http://arxiv.org/abs/1212.5701">paper</a>.
+ */
+public class AdaDelta extends Optimizer {
+
+  public static final String ACCUMULATOR = "accum";
+  public static final String ACCUMULATOR_UPDATE = "accum_update";
+
+  private final float learningRate;
+
+  private final float rho;
+
+  private final float epsilon;
```

Comment:
I don't think we need to enforce it but just FYI, pretty much all TF classes declare their members in this order, including both fields and methods: `public`, `protected`, `default`, `private`

Reply:
Do you mean all the public methods & fields, then protected etc, or all the public methods, protected methods, ... then public fields, protected fields etc?